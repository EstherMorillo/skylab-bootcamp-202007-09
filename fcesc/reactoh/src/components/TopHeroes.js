import React, {useState, useEffect} from 'react';
import './../css/paginatedList.css';
import { Link } from 'react-router-dom';
import HeroStore from './../stores/heroStore';
import { loadPaginatedHeroes } from './../actions/heroActions';
import * as ROUTES from './../config/routes';

function TopHeroes(){
    const [currentHeroes, setHeroes] = useState([]);
    const [page] = useState(0);
    const [itemsPerPage] = useState(5);

    useEffect(()=>{
        HeroStore.addChangeListener(onChange);
        if( currentHeroes.length === 0){ loadPaginatedHeroes(page, itemsPerPage) }
        return ()=>{HeroStore.removeChangeListener(onChange);}
    }, [currentHeroes.length, page, itemsPerPage]);

    function onChange(){
        setHeroes(HeroStore.getPaginatedHeroes(page, itemsPerPage)); 
    }

    return (
        <div className="paginatedList__container">
            <ul>
                {currentHeroes.map((hero)=>(
                    <li key={hero.id}>
                        <Link to={ROUTES.HERO_DETAIL_ROOT + '/' + hero.id}>{hero.id}:{hero.name}</Link>
                    </li>
                ))}
            </ul>
        </div>
    );
}

export default TopHeroes;