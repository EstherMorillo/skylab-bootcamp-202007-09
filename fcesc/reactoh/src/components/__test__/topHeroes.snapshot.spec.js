import React from 'react';
import renderer from 'react-test-renderer';
import TopHeroes from './../TopHeroes';
import { BrowserRouter as Router } from 'react-router-dom';

const tree = renderer.create(<Router><TopHeroes /></Router>).toJSON();

describe('TopHeroes snapshot', () => {
    test('should match', () => {
        expect(tree).toMatchSnapshot();
    })
})
