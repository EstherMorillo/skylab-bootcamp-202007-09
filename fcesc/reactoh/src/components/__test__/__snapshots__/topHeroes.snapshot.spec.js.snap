// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopHeroes snapshot should match 1`] = `
<div
  className="paginatedList__container"
>
  <ul />
</div>
`;
